artifactory/services/delete.go

```diff
@@ -210,7 +210,7 @@ func removeNotToBeDeletedDirs(specFile *utils.CommonParams, ds *DeleteService, d
 	if err != nil {
 		return nil, err
 	}
-	bufferFiles, err := utils.FilterCandidateToBeDeleted(deleteCandidates, resultWriter, "folder")
+	bufferFiles, err := utils.FilterCandidateToBeDeleted(deleteCandidates, resultWriter, string(utils.Folder))
```

Comment:
Let's use ResultItemType in FilterCandidateToBeDeleted too. This will eliminate many castings.
```suggestion
	bufferFiles, err := utils.FilterCandidateToBeDeleted(deleteCandidates, resultWriter, utils.Folder)
```